bot/exts/utilities/rfc.py

```diff
@@ -0,0 +1,102 @@
+import datetime
+import logging
+
+import pydantic
+from discord import Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+logger = logging.getLogger(__name__)
+
+BASE_URL = "https://datatracker.ietf.org/doc/rfc{rfc_id}/doc.json"
+
+
+class RfcDocument(pydantic.BaseModel):
+    """Represents an RFC document."""
+
+    title: str
+    description: str
+    revisions: str
+    created: datetime.datetime
+
+
+class Rfc(commands.Cog):
+    """Retrieves RFCs by their ID."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.cache: dict[int, RfcDocument] = {}
+
+    async def retrieve_data(self, rfc_id: int) -> RfcDocument | None:
+        """Retrieves the RFC from the cache or API, and adds to the cache if it does not exist."""
+        if rfc_id in self.cache:
+            return self.cache[rfc_id]
+
+        async with self.bot.http_session.get(BASE_URL.format(rfc_id=rfc_id)) as resp:
+            if resp.status != 200:
+                return None
+
+            data = await resp.json()
+
+        description = (
+            data["abstract"] or f"[Link](https://datatracker.ietf.org/doc/rfc{rfc_id})"
```

Comment:
I think we should link to the rfc even if the abstract is included, in case someone wants to read further.